src/main/java/games/strategy/engine/data/GameDataMemento.java

```diff
@@ -0,0 +1,168 @@
+package games.strategy.engine.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import games.strategy.util.Version;
+import games.strategy.util.memento.MementoExporter;
+import games.strategy.util.memento.MementoImportException;
+import games.strategy.util.memento.MementoImporter;
+import games.strategy.util.memento.PropertyBagMementoExporter;
+import games.strategy.util.memento.PropertyBagMementoImporter;
+
+/**
+ * Provides factory methods for creating objects that can import mementos to and export mementos from game data.
+ */
+public final class GameDataMemento {
+  @VisibleForTesting
+  static final long CURRENT_SCHEMA_VERSION = 1L;
+
+  @VisibleForTesting
+  static final String SCHEMA_ID = "application/x.triplea.game-data-memento";
+
+  /**
+   * An immutable collection of default memento export options.
+   *
+   * <p>
+   * The key is the option name. The value is the option value.
+   * </p>
+   */
+  public static final Map<ExportOptionName, Object> DEFAULT_EXPORT_OPTIONS_BY_NAME = newDefaultExportOptionsByName();
+
+  private GameDataMemento() {}
+
+  @VisibleForTesting
+  static final class PropertyNames {
+    static final String NAME = "name";
+    static final String VERSION = "version";
+
+    private PropertyNames() {}
+  }
+
+  private static Map<ExportOptionName, Object> newDefaultExportOptionsByName() {
+    return Maps.immutableEnumMap(ImmutableMap.of(
+        ExportOptionName.EXCLUDE_DELEGATES, false));
+  }
+
+  /**
+   * Creates a new game data memento exporter using the default export options.
+   *
+   * @return A new game data memento exporter; never {@code null}.
+   */
+  public static MementoExporter<GameData> newExporter() {
+    return newExporter(DEFAULT_EXPORT_OPTIONS_BY_NAME);
+  }
+
+  /**
+   * Creates a new game data memento exporter using the specified export options.
+   *
+   * @param optionsByName The memento export options; must not be {@code null}. The key is the option name. The value is
+   *        the option value.
+   *
+   * @return A new game data memento exporter; never {@code null}.
+   */
+  public static MementoExporter<GameData> newExporter(final Map<ExportOptionName, Object> optionsByName) {
+    checkNotNull(optionsByName);
+
+    return new PropertyBagMementoExporter<>(SCHEMA_ID, CURRENT_SCHEMA_VERSION, new ExportHandlers(optionsByName));
+  }
+
+  /**
+   * Names the options that can be specified when creating a new game data memento exporter.
+   */
+  public static enum ExportOptionName {
+    /**
+     * Indicates delegates should be excluded from the memento (value type: {@code Boolean}).
+     */
+    EXCLUDE_DELEGATES
```

Comment:
How is this used? A bit curious what is meant by a delegate too. TripleA had a pretty custom meaning of it, since there are two or there other common usages that have come about it.